include/boost/math/distributions/von_mises.hpp

```diff
@@ -0,0 +1,842 @@
+//  Copyright John Maddock 2006, 2007.
+//  Copyright Paul A. Bristow 2006, 2007.
+
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_STATS_VON_MISES_HPP
+#define BOOST_STATS_VON_MISES_HPP
+
+// https://en.wikipedia.org/wiki/Von_Mises_distribution
+// From MathWorld--A Wolfram Web Resource.
+// http://mathworld.wolfram.com/VonMisesDistribution.html
+
+#include <boost/math/distributions/fwd.hpp>
+#include <boost/math/distributions/complement.hpp>
+#include <boost/math/distributions/detail/common_error_handling.hpp>
+#include <boost/math/special_functions/bessel.hpp> // for besseli0 and besseli1
+#include <boost/math/special_functions/erf.hpp> // for erf
+#include <boost/math/tools/roots.hpp>
+
+#include <utility>
+
+namespace boost{ namespace math{
+
+template <class RealType = double, class Policy = policies::policy<> >
+class von_mises_distribution
+{
+public:
+  typedef RealType value_type;
+  typedef Policy policy_type;
+
+  von_mises_distribution(RealType l_mean = 0, RealType concentration = 1)
+    : m_mean(l_mean), m_concentration(concentration)
+  { // Default is a 'standard' von Mises distribution vM01.
+    static const char* function = "boost::math::von_mises_distribution<%1%>::von_mises_distribution";
+
+    RealType result;
+    detail::check_positive_x(function, concentration, &result, Policy());
+    detail::check_angle(function, l_mean, &result, Policy());
+  }
+
+  RealType mean() const
+  { // alias for location.
+    return m_mean;
+  }
+
+  RealType concentration() const
+  { // alias for scale.
+    return m_concentration;
+  }
+
+  // Synonyms, provided to allow generic use of find_location and find_scale.
+  RealType location() const
+  { // location.
+    return m_mean;
+  }
+  RealType scale() const
+  { // scale.
+    return m_concentration;
+  }
+
+private:
+  //
+  // Data members:
+  //
+  RealType m_mean;  // distribution mean or location.
+  RealType m_concentration;    // distribution standard deviation or scale.
+}; // class von_mises_distribution
+
+typedef von_mises_distribution<double> von_mises;
+
+#ifdef BOOST_MSVC
+#pragma warning(push)
+#pragma warning(disable:4127)
```

Comment:
Could you add a comment on why this warning should be disabled?

Reply:
MSVC used to warn on anything that was constant.

// C 4127 "conditional expression is constant"

They have recently stopped doing this, at last, thank goodness!

https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4127?view=vs-2019

But we still test antediluvian versions of MSVC, so still worth keeping to reduce clutter?